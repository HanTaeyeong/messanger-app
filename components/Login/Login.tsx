import React from 'react'

export default function Login() {
  return (
    <div>
      <input className='login-input'></input>
      <button className='login-button'>Sign in</button>
    </div>
  )
}
