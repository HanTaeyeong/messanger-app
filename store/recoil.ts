import { atom, selector, useRecoilState } from 'recoil';

export const weekList: String[] = ['Sunday', 'Monday', 'Tuesday', 'Wendsday', 'Thursday', 'Fryday', 'Saturday']

export const testDate = atom<Number>({
    key: 'testDate',
    default: 0,
});

export const testWeekName = selector<String>({
    key: 'testWeekName',
    get: ({ get }) => weekList[(+get(testDate) % 7) | 0] || 'Sunday',
    set: ({ set }) => { }
});
