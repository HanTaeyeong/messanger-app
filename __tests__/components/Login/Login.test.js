import React from 'react'

import Login from '../../../components/Login/Login'
import TestRenderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil'

describe('Login Component test', () => {
    let testComponent;

    beforeEach(() => {
        testComponent = TestRenderer.create(
            <RecoilRoot>
                <Login />
            </RecoilRoot>)
    })

    test('renders well', () => {
        expect(testComponent).toBeTruthy()
        expect(testComponent.toJSON()).toMatchSnapshot();
    })
    test('head describes page well', () => {
        expect(testComponent.root.findByProps({ className: "login-button" })).toBeTruthy()
    })
})
