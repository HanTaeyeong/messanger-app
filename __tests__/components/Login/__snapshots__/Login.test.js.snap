// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component test renders well 1`] = `
<div>
  <form>
    <input
      className="login-input"
      name="ID"
      onChange={[Function]}
      placeholder="ID"
    />
    <input
      className="login-input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
    />
    <button
      className="login-button"
      onClick={[Function]}
    >
      Sign in
    </button>
  </form>
</div>
`;
